File: backend/app/schemas/prompt.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class PromptTemplateBase(BaseModel):
    name: str = Field(..., description="模板名称")
    content: str = Field(..., description="提示词内容")
    description: str | None = Field(None, description="模板描述")
    is_active: bool = Field(True, description="是否激活")
    is_system: bool = Field(False, description="是否为系统内置")
    category: str = Field("general", description="分类")

class PromptTemplateCreate(PromptTemplateBase):
    pass

class PromptTemplateUpdate(BaseModel):
    name: str | None = None
    content: str | None = None
    description: str | None = None
    is_active: bool | None = None
    category: str | None = None

class PromptTemplateResponse(PromptTemplateBase):
    id: int
    created_at: datetime
    updated_at: datetime | None = None
    creator_id: int | None = None
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/prompt.ts
import request from '@/utils/request'
import type { AxiosResponse } from 'axios'

export interface PromptTemplate {
  id: number
  name: string
  content: string
  description?: string
  is_active: boolean
  is_system: boolean
  category: string
  created_at: string
  updated_at?: string
  creator_id?: number
}

export const getPrompts = async (params?: {
  skip?: number
  limit?: number
  category?: string
}): Promise<AxiosResponse<PromptTemplate[]>> => {
  return request.get('/prompts/', { params })
}

export const createPrompt = async (data: Partial<PromptTemplate>): Promise<AxiosResponse<PromptTemplate>> => {
  return request.post('/prompts/', data)
}

export const updatePrompt = async (id: number, data: Partial<PromptTemplate>): Promise<AxiosResponse<PromptTemplate>> => {
  return request.put(`/prompts/${id}`, data)
}

export const deletePrompt = async (id: number): Promise<void> => {
  return request.delete(`/prompts/${id}`)
}

export const seedDefaultPrompts = async (): Promise<void> => {
  return request.post('/prompts/seed')
}
